### PR TITLE
Add prepare flag to cli

### PR DESCRIPTION
**Pull request by nmlinaric**
## Description
Add `--prepare` flag for logging calldata

Closes: #178 

## How Has This Been Tested? Testing details.
Unit tests + manual testing

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/evm/calls/transactor/transact.go [hunk 1]
```diff
@@ -83,3 +89,8 @@ func (t *signAndSendTransactor) Transact(to *common.Address, data []byte, opts T
 	}
 	return &h, nil
 }
+
+func (t *prepareTransactor) Transact(to *common.Address, data []byte, opts TransactOptions) (*common.Hash, error) {
+	log.Info().Msgf("Command calldata: %+v", data)
```
Comment (mpetrun5) [inline chains/evm/calls/transactor/transact.go:94]: Probably we should use some pretty print here and also print out to address

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

File: chains/evm/calls/transactor/transact.go (hunk 1 above)
Comment (P1sar) [inline chains/evm/calls/transactor/transact.go:94]: Please take a look at the issue in more time. And think what is the purpose of this task, and HOW printed data will be used

File: chains/evm/cli/initialize/initialize.go [hunk 2]
```diff
@@ -2,13 +2,14 @@ package initialize
...
 	"github.com/ChainSafe/chainbridge-core/chains/evm/calls"
 	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/evmclient"
 	evmgaspricer "github.com/ChainSafe/chainbridge-core/chains/evm/calls/evmgaspricer"
 	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/transactor"
 	"github.com/ChainSafe/chainbridge-core/crypto/secp256k1"
 	"github.com/rs/zerolog/log"
-	"math/big"
 )
 
 func InitializeClient(
```
Comment (P1sar) [inline chains/evm/cli/initialize/initialize.go:15]: Please add godoc for this method, make sure that purpose of prepare argument is explicitly described

File: chains/evm/calls/transactor/transact.go [hunk 3]
```diff
@@ -48,6 +50,10 @@ func NewSignAndSendTransactor(txFabric calls.TxFabric, gasPriceClient calls.GasP
 	}
 }
 
+func NewPrepareTransactor() Transactor {
```
Comment (P1sar) [inline chains/evm/calls/transactor/transact.go:53]: Add godoc for this function

File: chains/evm/calls/transactor/transact.go [hunk 4]
```diff
@@ -83,3 +89,8 @@ func (t *signAndSendTransactor) Transact(to *common.Address, data []byte, opts T
 	}
 	return &h, nil
 }
+
+func (t *prepareTransactor) Transact(to *common.Address, data []byte, opts TransactOptions) (*common.Hash, error) {
```
Comment (P1sar) [inline chains/evm/calls/transactor/transact.go:93]: add godoc for this function

**Review by P1sar — CHANGES_REQUESTED** [review]

**Comment by P1sar** [comment]
This flag should also be documented in README CLI section

**Review by P1sar — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -74,6 +74,7 @@ Flags:
       --network uint                ID of the Network
       --private-key string          Private key to use
       --url string                  URL of the node to receive RPC calls (default "ws://localhost:8545")
+      --prepare                     Generate calldata for contract calls
```
Comment (mpetrun5) [inline README.md:77]: Maybe also add that it doesn't require private key and it doesn't send transaction

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

**Review by MakMuftic — APPROVED** [review]

**Pull request merged**